File: src/main/Java/boj_step8/boj2745.java
import java.io.*;
import java.util.StringTokenizer;

public class boj2745 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        String n = tokenizer.nextToken();
        int b = Integer.parseInt(tokenizer.nextToken());
        writer.append(String.valueOf(Integer.parseInt(n, b)));
        writer.close();
    }
}

File: src/main/kotlin/boj2745.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val (n, b) = r.readLine().split(" ").map { it }
    w.write(n.toBigInteger(b.toInt()).toString())
    w.close()
}